syft/execution/type_wrapper.py

```diff
@@ -0,0 +1,261 @@
+import syft as sy
+from syft.workers.abstract import AbstractWorker
+from warnings import warn
+from typing import Union
+from syft_proto.execution.v1.type_wrapper_pb2 import NestedTypeWrapper as NestedTypeWrapperPB
+from syft_proto.execution.v1.type_wrapper_pb2 import ClassType as ClassTypePB
+
+class NestedTypeWrapper:
```

Comment:
Is the purpose of this class to capture the types of input args? If so, there's probably a more descriptive name for it. Maybe something like `InputTypeDescriptor`?